src/node_buffer.cc

```diff
@@ -19,7 +19,11 @@
   } while (0)
 
 #define ARGS_THIS(argT)                                                     \
-  Local<Object> obj = argT;                                                 \
+  Handle<Value> _obj = argT;                                                \
+  if (!HasInstance(_obj)) {                                                 \
+    return env->ThrowTypeError("must be called on Buffer instance");        \
```

Comment:
> um. do we just want to hold off on this since Buffer has to be completely rewritten anyway since 4.4 will break everything?

If it's gona be re-written might as well re-write it to use ArrayBuffers by the sounds of it. :S